arcgis-ios-sdk-samples/Maps/Manage operational layers/ManageMapLayersViewController.swift

```diff
@@ -15,51 +15,73 @@
-        self.map = AGSMap(basemap: .topographic())
+        let map = AGSMap(basemap: .topographic())
+        map.initialViewpoint = AGSViewpoint(center: AGSPoint(x: -133e5, y: 45e5, spatialReference: .webMercator()), scale: 2e7)
         
-        let imageLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Elevation/WorldElevations/MapServer")!)
-        self.map.operationalLayers.add(imageLayer)
+        let elevationImageLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Elevation/WorldElevations/MapServer")!)
+        let censusTiledLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Census/MapServer")!)
         
-        let tiledLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Census/MapServer")!)
-        self.map.operationalLayers.add(tiledLayer)
-
-        self.deletedLayers = [AGSLayer]()
+        allLayers = [elevationImageLayer, censusTiledLayer]
         
-        //add the source code button item to the right of navigation bar
-        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["ManageMapLayersViewController", "MMLLayersViewController"]
+        // load all the layers into the map to start
+        map.operationalLayers.addObjects(from: allLayers)
         
-        self.mapView.map = map
-        self.mapView.setViewpoint(AGSViewpoint(center: AGSPoint(x: -133e5, y: 45e5, spatialReference: AGSSpatialReference(wkid: 3857)), scale: 2e7))
+        mapView.map = map
     }
     
     //MARK: - Navigation
     
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        if segue.identifier == "LayersSegue" {
-            
-            let navigationController = segue.destination as! UINavigationController
-            let controller = navigationController.viewControllers[0] as! MMLLayersViewController
-            controller.layers = self.map.operationalLayers
-            controller.deletedLayers = self.deletedLayers
-            controller.preferredContentSize = CGSize(width: 300, height: 300)
-            controller.delegate = self
+        if let navController = segue.destination as? UINavigationController,
+            let controller = navController.viewControllers.first as? MMLLayersViewController {
+            // convert and assign the map's operational layers as a Swift array
+            controller.operationalLayers = mapView.map?.operationalLayers as? [AGSLayer] ?? []
+            controller.removedLayers = removedLayers
+            controller.onChange = {[weak self] layers in
+                guard let map = self?.mapView.map else {
+                    return
+                }
+                // replace the map's operational layers with those from the view controller
+                map.operationalLayers.removeAllObjects()
```

Comment:
Dealing with an optional `NSMutableArray` is not great, but I'm not sure this is any better. Removing all layers and re-adding them on each change does not seem like a good idea. Wouldn't that cause all layers to be redrawn, not just the one that was added or removed?